ga4gh/datamodel/reads.py

```diff
@@ -289,6 +300,23 @@ def getNumUnalignedReads(self):
     def getPrograms(self):
         return self._programs
 
+    def getReadAlignments(self, reference, start, end):
+        """
+        Returns an iterator over all reads in a ReadGroupSet
+        """
+        samFile = self.getFileHandle(self._samFilePath)
+        referenceName = reference.getLocalId().encode()
+        start, end = self.sanitizeAlignmentFileFetch(start, end)
+        readAlignments = samFile.fetch(referenceName, start, end)
+        syntheticReadGroup = HtslibReadGroup(self, 'synthetic')
```

Comment:
Yes, that is the intention.  What refactor do you have in mind?  A mixin class that has getReadAlignments as a method?